### PR TITLE
Integrate llvm-project at 5cfc22cafe3f and bump dependencies

### PR DESCRIPTION
**Pull request by KoolJBlack**
LLVM_COMMIT=”5cfc22cafe3f2465e0bb324f8daba82ffcabd0df” MHLO_COMMIT=”7d965ddaad2f15278526d813d1b27b3a6ec3207a” TF_COMMIT=”2358d6bfee56c96924f0a2d85c7792e877de6ce2”

### CODE REVIEW

**Review by antiagainst — APPROVED** [review]

File: tests/e2e/matmul/BUILD
Comment (MaheshRavishankar) [inline tests/e2e/matmul/BUILD:261]: The split-k test failing can have downstream effects. Maybe we can dig into this a bit further before landing?

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]

**Comment by MaheshRavishankar** [comment]
Just added the benchmark labels to see if there are any failures on CUDA benchmarks... (though I think they are run anyway with the CI)

**Comment by MaheshRavishankar** [comment]
@KoolJBlack, @ThomasRaoux and @antiagainst  you guys know how critical the failure of split-k is. Ill let you guys decide if this integrate should land or not

**Comment by MaheshRavishankar** [comment]
Oh I didnt see this was on auto-merge... sigh....

**Pull request merged**
